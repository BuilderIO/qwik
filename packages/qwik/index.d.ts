// re-export to make TS happy when not using nodenext import resolution
export * from './src/core/index.ts'
