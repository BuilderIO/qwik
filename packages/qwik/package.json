{
  "name": "@builder.io/qwik",
  "description": "An Open-Source sub-framework designed with a focus on server-side-rendering, lazy-loading, and styling/animation.",
  "version": "1.5.1",
  "annotation": "This package.json is for internal use in the monorepo, the build actually makes a new package.json for the published package via scripts/package-json.ts",
  "bin": "./qwik-cli.cjs",
  "bugs": "https://github.com/BuilderIO/qwik/issues",
  "contributors": [
    {
      "name": "MiÅ¡ko Hevery",
      "email": "misko@hevery.com",
      "url": "https://github.com/BuilderIO/qwik"
    },
    {
      "name": "Adam Bradley",
      "email": "adam@builder.io",
      "url": "https://twitter.com/adamdbradley"
    },
    {
      "name": "Manu Mtz.-Almeida",
      "email": "manu@builder.io",
      "url": "https://twitter.com/manucorporat"
    }
  ],
  "dependencies": {
    "csstype": "^3.1.3",
    "vite": "^5.1.6"
  },
  "devDependencies": {
    "@builder.io/qwik": "workspace:^",
    "@builder.io/qwik-dom": "workspace:^",
    "image-size": "^1.0.2",
    "kleur": "4.1.5"
  },
  "engines": {
    "node": ">=16.8.0 <18.0.0 || >=18.11"
  },
  "exports": {
    ".": "./src/core/index.ts",
    "./cli": "./src/cli/index.ts",
    "./jsx-runtime": "./src/jsx-runtime/index.ts",
    "./jsx-dev-runtime": "./src/jsx-runtime/index.ts",
    "./build": "./src/build/index.ts",
    "./optimizer.cjs": "./dist/optimizer.cjs",
    "./optimizer.mjs": "./dist/optimizer.mjs",
    "./optimizer.broken": "./src/optimizer/src/index.ts",
    "./optimizer": "./dist/optimizer.mjs",
    "./server.cjs": "./dist/server.cjs",
    "./server.mjs": "./dist/server.mjs",
    "./server": "./src/server/index.ts",
    "./testing": "./src/testing/index.ts",
    "./qwikloader.js": "./dist/qwikloader.js",
    "./qwikloader.debug.js": "./dist/qwikloader.debug.js",
    "./qwik-prefetch.js": "./dist/qwik-prefetch.js",
    "./qwik-prefetch.debug.js": "./dist/qwik-prefetch.debug.js",
    "./package.json": "./package.json"
  },
  "exports_annotation": "We use the build for the optimizer because esbuild doesn't like the html?raw imports in the server plugin and it's only used in the vite configs",
  "files": [
    "bindings/qwik.darwin-x64.node",
    "bindings/qwik.darwin-arm64.node",
    "bindings/qwik.win32-x64-msvc.node",
    "bindings/qwik_wasm_bg.wasm",
    "bindings/qwik.wasm.cjs",
    "bindings/qwik.wasm.mjs",
    "build/index.cjs",
    "build/index.cjs.map",
    "build/index.mjs",
    "build/index.mjs.map",
    "build/index.dev.mjs",
    "build/index.dev.mjs.map",
    "build/index.prod.mjs",
    "build/index.prod.mjs.map",
    "build/index.dev.cjs",
    "build/index.dev.cjs.map",
    "build/index.prod.cjs",
    "build/index.prod.cjs.map",
    "build/index.d.ts",
    "build/package.json",
    "cli.cjs",
    "core/package.json",
    "core.cjs",
    "core.cjs.map",
    "core.min.mjs",
    "core.mjs",
    "core.mjs.map",
    "core.prod.cjs",
    "core.prod.mjs",
    "core.d.ts",
    "jsx-dev-runtime/package.json",
    "jsx-runtime/index.d.ts",
    "jsx-runtime/package.json",
    "jsx-runtime.cjs",
    "jsx-runtime.cjs.map",
    "jsx-runtime.mjs",
    "jsx-runtime.mjs.map",
    "jsx-runtime.d.ts",
    "loader/index.cjs",
    "loader/index.mjs",
    "loader/index.d.ts",
    "loader/package.json",
    "optimizer/package.json",
    "optimizer.cjs",
    "optimizer.mjs",
    "optimizer.d.ts",
    "prefetch/index.cjs",
    "prefetch/index.d.ts",
    "prefetch/index.mjs",
    "prefetch/package.json",
    "qwik-prefetch.debug.js",
    "qwik-prefetch.js",
    "qwikloader.js",
    "qwikloader.debug.js",
    "qwik-cli.cjs",
    "server.cjs",
    "server.mjs",
    "server.d.ts",
    "server-modules.d.ts",
    "server/package.json",
    "starters/adapters",
    "starters/features",
    "templates",
    "testing/index.cjs",
    "testing/index.mjs",
    "testing/index.d.ts",
    "testing/package.json"
  ],
  "homepage": "https://qwik.dev/",
  "keywords": [
    "Builder.io",
    "framework",
    "generator",
    "prerender",
    "server-side-render",
    "ssg",
    "ssr",
    "static-site",
    "static-site-generator",
    "webapp",
    "website"
  ],
  "license": "MIT",
  "main": "./src/index.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/BuilderIO/qwik.git",
    "directory": "packages/qwik"
  },
  "sideEffects": false,
  "type": "module"
}
