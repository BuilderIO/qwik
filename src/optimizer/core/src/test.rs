extern crate insta;
use super::*;

#[test]
fn example_1() {
    test_input(
        "test.tsx",
        r#"
const Header = qComponent({
  "onMount": qHook(() => { console.log("mount") }),
  onRender: qHook(() => {
    return (
      <div onClick={qHook((ctx) => console.log(ctx))}/>
    );
  })
});
    "#,
    )
}

#[test]
fn example_2() {
    test_input(
        "test.tsx",
        r#"
export const Header = qComponent({
  "onMount": qHook(() => { console.log("mount") }),
  onRender: qHook(() => {
    return (
      <div onClick={qHook((ctx) => console.log(ctx))}/>
    );
  })
});
    "#,
    )
}

#[test]
fn example_3() {
    test_input(
        "test.tsx",
        r#"
export const App = () => {
    const Header = qComponent({
        "onMount": qHook(() => { console.log("mount") }),
        onRender: qHook(() => {
            return (
            <div onClick={qHook((ctx) => console.log(ctx))}/>
            );
        })
    });
    return Header;
});
    "#,
    )
}

#[test]
fn example_4() {
    test_input(
        "test.tsx",
        r#"
export const Header = qComponent({
    onRender: qHook(() => {
        return (
            <>
                <div onClick={qHook((ctx) => console.log("1"))}/>
                <div onClick={qHook((ctx) => console.log("2"))}/>
            </>
        );
    })
});
    "#,
    )
}


#[test]
fn example_5() {
    test_input(
        "test.tsx",
        r#"
export const sym1 = qHook((ctx) => console.log("1"));
    "#,
    )
}

#[test]
fn example_6() {
    test_input(
        "test.tsx",
        r#"
import {qHook} from '@builderio/qwik';


const Header = qComponent({
    "onMount": qHook(() => { console.log("mount") }),
    onRender: qHook(() => {
      return (
        <div onClick={qHook((ctx) => console.log(ctx))}/>
      );
    })
  });

const App = qComponent({
    onRender: qHook(() => {
        return (
        <Header/>
        );
    })
});"#,
    )
}


#[test]
fn optimize_fixture_1() {
    test_fixture("");
}

fn test_fixture(folder: &str) {
    let res = transform_workdir(&FSConfig {
        project_root: folder.to_string(),
        source_maps: true,
        minify: false,
        transpile: false,
    });
    insta::assert_debug_snapshot!(res);
}

fn test_input(filename: &str, code: &str) {
    let res = transform(Config {
        code: code.as_bytes().to_vec(),
        filename: filename.to_string(),
        source_maps: true,
        minify: false,
        transpile: false,
    });
    match res {
        Ok(v) => {
            let s = v.to_string();
            insta::assert_display_snapshot!(if let Some(code) = s.code { code } else { "".to_string() });
            insta::assert_display_snapshot!(if let Some(map) = s.map { map } else { "".to_string() });
            insta::assert_debug_snapshot!(v.diagnostics);
        }
        Err(err) => {
            insta::assert_display_snapshot!(err);
        }
    }
}
