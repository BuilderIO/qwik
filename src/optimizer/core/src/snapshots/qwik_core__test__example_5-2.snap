---
source: src/test.rs
expression: "if let Some(map) = s.map { map } else { \"\".to_string() }"

---
{"version":3,"sources":["test.tsx"],"names":[],"mappings":"sBACoB,KAAK,EAAE,GAAG,GAAK,OAAO,CAAC,GAAG,CAAC,CAAG;;AAAlD,MAAM,CAAC,KAAK,CAAC,IAAI"}
